src/client/Session.cpp

```diff
@@ -22,7 +23,37 @@ void Session::asyncExecute(const std::string &stmt, ExecuteCallback cb) {
 
 ExecutionResponse Session::executeWithParameter(
     const std::string &stmt, const std::unordered_map<std::string, Value> &parameters) {
-  return ExecutionResponse(conn_.executeWithParameter(sessionId_, stmt, parameters));
+  if (connectionIsBroken_ && retryConnect_) {
```

Comment:
Why need this block?

Reply:
This implementation is from [java client](https://github.com/vesoft-inc/nebula-java/blob/b703e03bb062b8fac8989ddc35f2b3243b4d3510/client/src/main/java/com/vesoft/nebula/client/graph/net/Session.java#L116), when doing a query, the connection maybe broken, so, we can retry the connection automatically to make sure the query can be successful.

Reply:
Got.